src/ProjectTemplates/Microsoft.Extensions.AI.Templates/Microsoft.Extensions.AI.Templates.csproj

```
@@ -20,20 +20,86 @@
     <EnableDefaultItems>false</EnableDefaultItems>
     <NoDefaultExcludes>true</NoDefaultExcludes>
     <NoPackageAnalysis>true</NoPackageAnalysis>
+
+    <!-- Use for incremental builds. When versions or dependencies of templates change, this file is updated and causes a re-build. -->
+    <_GeneratedContentPropertiesHashFile>$(IntermediateOutputPath)$(MSBuildProjectName).content.g.cache</_GeneratedContentPropertiesHashFile>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Templating" Version="$(MicrosoftDotNetBuildTasksTemplatingVersion)" AllowExplicitReference="true" PrivateAssets="All" IsImplicitlyDefined="true" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <!-- Specify package version variables used in template content. -->
+    <_GeneratedContentPackageVersionProperties>
+      OllamaSharpVersion=$(OllamaSharpVersion);
+      OpenAIVersion=$(OpenAIVersion);
+      AzureAIProjectsVersion=$(AzureAIProjectsVersion);
+      AzureAIOpenAIVersion=$(AzureAIOpenAIVersion);
+      AzureIdentityVersion=$(AzureIdentityVersion);
+      MicrosoftEntityFrameworkCoreSqliteVersion=$(MicrosoftEntityFrameworkCoreSqliteVersion);
+      MicrosoftSemanticKernelCoreVersion=$(MicrosoftSemanticKernelCoreVersion);
+      PdfPigVersion=$(PdfPigVersion);
+      SystemLinqAsyncVersion=$(SystemLinqAsyncVersion);
+      AzureSearchDocumentsVersion=$(AzureSearchDocumentsVersion);
+      MicrosoftSemanticKernelConnectorsAzureAISearchVersion=$(MicrosoftSemanticKernelConnectorsAzureAISearchVersion);
+    </_GeneratedContentPackageVersionProperties>
   </PropertyGroup>
 
   <ItemGroup>
-    <Content Include="src\ChatWithCustomData\**\*" Exclude="**\bin\**;**\obj\**;**\node_modules\**;**\*.user;**\*.csproj.in;**\*.out.js;**\*.generated.css;**\package-lock.json" />
+    <_GeneratedContent
+      Include="src\ChatWithCustomData\ChatWithCustomData.Web\ChatWithCustomData.Web.csproj.in"
+      OutputPath="src\ChatWithCustomData\ChatWithCustomData.Web\ChatWithCustomData.Web.csproj" />
+    <_GeneratedContent
+      Include="src\ChatWithCustomData\ChatWithCustomData.Web\Directory.Build.targets.in"
+      OutputPath="src\ChatWithCustomData\ChatWithCustomData.Web\Directory.Build.targets" />
+    <_GeneratedContent
+      Include="src\ChatWithCustomData\ChatWithCustomData.Web\NuGet.config.in"
+      OutputPath="src\ChatWithCustomData\ChatWithCustomData.Web\NuGet.config" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Include="src\ChatWithCustomData\**\*" Exclude="**\bin\**;**\obj\**;**\node_modules\**;**\*.user;**\*.in;**\*.out.js;**\*.generated.css;**\package-lock.json;**\NuGet.Config;**\Directory.Build.targets" />
```

```suggestion
    <Content Include="src\ChatWithCustomData\**\*"
             Exclude="**\bin\**;**\obj\**;**\node_modules\**;**\*.user;**\*.in;**\*.out.js;**\*.generated.css;**\package-lock.json;**\NuGet.Config;**\Directory.Build.targets" />
```

And maybe even split the content into individual lines:

```suggestion
    <Content Include="src\ChatWithCustomData\**\*"
             Exclude="**\bin\**;
                      **\obj\**;
                      **\node_modules\**;
                      **\*.user;
                      **\*.in;
                      **\*.out.js;
                      **\*.generated.css;
                      **\package-lock.json;
                      **\NuGet.Config;
                      **\Directory.Build.targets" />
```